libs/s25main/desktops/dskGameInterface.cpp

```diff
@@ -409,7 +409,7 @@ void dskGameInterface::Msg_PaintAfter()
     DrawPoint iconPos(VIDEODRIVER.GetRenderSize().x - 56, 32);
 
     // Draw cheating indicator icon (WINTER)
-    if(isCheatModeOn)
+    if(world.IsCheatModeOn())
```

Comment:
IMO cheats are not part of the world but rather belong to the interface. Or is there a reason for moving them down? I'd make an instance of `Cheats` in this class instead which also avoids going through `game->world_.GetCheats` and the inconsistent access here, i.e. why isn't this using `GetCheats`?

Reply:
There is nothing wrong with cheats modifying the world. They also don't need the gameinterface as that would be backwards: The GameInterface **has** a cheats and world. The cheats **use** the world (modify it)

So my idea was to pass the world into the cheats class (controller->model pattern) where needed.

But I see you need the cheats in the GameWorldViewer. Maybe they are better of there. That could break the cycle: GWV has cheats and world-reference, cheats has world-reference. That could even be just a reference in GWV where dskGameInterface has the actual class (and possibly the handlers/tracker)

Actually I caused some confusion here: The interface is the desktop, not the GameInterface class itself. I guess that would work then.

Reply:
I move the Cheats instance from GameWorldBase to dskGameInterface but retained the helper functions in GameWorldBase because they are convenienet there. Resolving - please unresolve if necessary, otherwise I think this can be merged.